bug 1531449 add note on master about restart with cloud outage

bobfuru opened the pull request:
Approved by SME, change made on master for CP to 4.1, 4.2. Relates to [PR 16068 ](https://github.com/openshift/openshift-docs/pull/16068/), where note was added to master-3 for 3.9, 3.10, 3.11. 
Content location changed from 3.x (architecture/core_concepts/pods_and_services.adoc)
/to 4.x (modules/nodes-pods-configuring-restart.adoc), so placed note in same text location.

[BZ 1531449](https://bugzilla.redhat.com/show_bug.cgi?id=1531449)

@openshift/team-documentation Peer Review Needed

bobfuru commented:
@derekwaynecarr please review and confirm whether it is possible to disable integration with a cloud provider in 4.x.

mburke5678 commented:
@bobfuru LGTM pending Derek's response to https://github.com/openshift/openshift-docs/pull/16158#issuecomment-519255025.
If supported, do we document how to disable integration?

bobfuru commented:
> @bobfuru LGTM pending Derek's response to [#16158 (comment)](https://github.com/openshift/openshift-docs/pull/16158#issuecomment-519255025).
> If supported, do we document how to disable integration?

Good question, @mburke5678 . I can't find where we do document this process. @kalexand-rh Do you know if we document how to disable integration anywhere?

vikram-redhat commented:
> > @bobfuru LGTM pending Derek's response to [#16158 (comment)](https://github.com/openshift/openshift-docs/pull/16158#issuecomment-519255025).
> > If supported, do we document how to disable integration?
> 
> Good question, @mburke5678 . I can't find where we do document this process. @kalexand-rh Do you know if we document how to disable integration anywhere?

@derekwaynecarr can confirm, but I don't think disabling integration is a simple process and unlikely to be documented anywhere.

kalexand-rh commented:
@vikram-redhat, if it's possible, it was never mentioned. I think that the cluster's tightly wedded to its kubelet and etcd, so I would honestly be surprised if this was valid for 4.x. Even if it's possible, it's not documented, I doubt that we should expose it in 4.x. That's why I wanted Derek to approve the addition.

kalexand-rh commented on modules/nodes-pods-configuring-restart.adoc:
```diff
@@ -43,6 +43,10 @@ restart, use a restart policy of `Never`.
 
+[NOTE]
+====
+Kubernetes architecture expects reliable endpoints from cloud providers. When a cloud provider is down, the kubelet prevents {product-title} from restarting. If your underlying cloud provider is unavailable, it may be helpful to disable integration with {product-title} and configure the kubelet as a bare-metal or no-cloud environment.
```
s/may/might
s/bare-metal/bare metal
Is it helpful, or will you be able to restart your cluster if you disable the integration?

To my knowledge, you can't change the configuration of a 4.x cluster to bare metal after you've provisioned it.

bobfuru replied:
@kalexand-rh I applied your first two edits to master-3 only [PR16068](https://github.com/openshift/openshift-docs/pull/16068/) while awaiting SME feedback for 4.x impact. 

Not sure the answer to your question ("Is it helpful"), so hope @derekwaynecarr can also confirm if this is accurate when he reviews.

derekwaynecarr replied:
@kalexand-rh i see no value in taking an existing cluster that was configured to use a cloud provider and transform it to disable the cloud provider.  make the underlying cloud environment reliable, or install a new cluster.  it's probably possible to do that in 3.x, but I am certain it was never tested, and depending on how the cluster is used, it would cause chaos.  i think its more accurate to say that "If the underling cloud provider endpoints are not reliable, do not install a cluster using cloud provider integration.  Install the cluster as if it was in a no-cloud environment.  It is not recommended to toggle cloud provider integration on or off in an installed cluster."

bobfuru commented:
@derekwaynecarr 
Bump. 

Kathryn points out that it is questionable whether one can disable integration in 4.x as we suggest they do in 3.9+, modified in [PR 16068](https://github.com/openshift/openshift-docs/pull/16068/).

bobfuru commented:
Updated note approved by SME, squashed commits on master for CP to 4.1, 4.2. Relates to [PR 16068](https://github.com/openshift/openshift-docs/pull/16068), where note was added to master-3 for 3.9, 3.10, 3.11.

[BZ 1531449](https://bugzilla.redhat.com/show_bug.cgi?id=1531449)

@openshift/team-documentation Peer Review Needed

adellape commented:
/cherrypick enterprise-4.1

adellape commented:
/cherrypick enterprise-4.2
